src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,336 @@
+            return getHazelcastCloudCluster(clusterId);
+        } catch (Exception e) {
+            throw new CloudException(String.format("Resume cluster with id %s is failed, Rc stack trace is: ", clusterId, e.getStackTrace()));
+        }
+    }
+
+    public void deleteHazelcastCloudCluster(String clusterId) throws CloudException {
+        try {
+            String query = String.format("{\"query\": \"mutation { deleteCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(responseBody);
+            waitForDeletedCluster(clusterId, TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            LOG.info(String.format("Cluster with id %s is deleted", clusterId));
+        } catch (Exception e) {
+            throw new CloudException(String.format("Delete hazelcast cloud cluster with id %s is failed, Rc stack trace is: %s", clusterId, e.getMessage()));
+        }
+    }
+
+    private Response prepareAndSendRequest(String query) throws InterruptedException, CloudException {
+        int retryCountForExceptionOfEndpoint = 0;
+        // Rarely server returns empty header, that is why a retry mechanism is added.
+        while(retryCountForExceptionOfEndpoint < 3)
+        {
+            try {
+                RequestBody body = RequestBody.create(JSON, query);
+                Request request = new Request.Builder()
+                        .url(HttpUrl.get(uri))
+                        .post(body)
+                        .header("Authorization", String.format("Bearer %s", bearerToken))
+                        .header("Content-Type", "application/json")
+                        .build();
+
+                call = client.newCall(request);
+                return call.execute();
+            }
+            catch(Exception e)
+            {
+                Log.warn(e.toString());
+            }
+            retryCountForExceptionOfEndpoint++;
+            TimeUnit.SECONDS.sleep(2);
+        }
+        throw new CloudException("Request cannot send successfully after 3 tries");
+    }
+
+    private String getTlsPassword(String clusterId) {
+        String requestUrl = String.format("%s/cluster/%s", baseUrl, clusterId);
+        try {
+            Request request = new Request.Builder()
+                    .url(requestUrl)
+                    .get()
+                    .header("Authorization", String.format("Bearer %s", bearerToken))
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            try {
+                return mapper.readTree(response.body().string()).get("tlsPassword").asText();
+            } catch (Exception e) {
+                throw new CloudException("Body is null for tlsPassword");
+            }
+        }
+        catch(Exception e)
+        {
+            LOG.warn(e.getMessage());
```

Comment:
Please throw exception. Fail fast.

Reply:
Sometimes it needs to return null instead of throwing an exception. For example; checking a deleted cluster. If a cluster is deleted and calls this method it will throw an exception but it is a normal behaviour, it shouldn't throw an exception. We can handle it with another way but we need to change a few other methods. I prefer to keep it like that

Reply:
Please put javadoc to describe why we should not throw here.